packages/strapi-connector-mongoose/lib/queries.js

```diff
@@ -469,13 +486,29 @@ module.exports = ({ model, strapi }) => {
 
   async function deleteMany(params) {
     if (params[model.primaryKey]) {
+      if (!utils.isMongoId(params[model.primaryKey])) {
+        return null;
+      }
+
       const entries = await find({ ...params, _limit: 1 });
       if (entries.length > 0) {
         return deleteOne(entries[0][model.primaryKey]);
       }
       return null;
     }
 
+    const { _where = [] } = params;
+
+    for (let i = 0; i < _where.length; i++) {
```

Comment:
I don't think we need this whole block since the checks you do are only on the `pk_in` op, and doesn't handle the nested where clauses. We'll add the support for those checks later on.

Reply:
Removed